modules/persistent-storage-csi-manila-share-access-rules.adoc

```diff
@@ -0,0 +1,107 @@
+* You have cluster administrator privileges.
+
+.Procedure
+
+To create a custom storage class with restricted share access:
+
+. Create a YAML file for your custom storage class:
++
+.custom-manila-storageclass.yaml
+[source,yaml]
+----
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: csi-manila-gold-restricted <1>
+provisioner: manila.csi.openstack.org
+parameters:
+  type: gold <2>
+  nfs-ShareClient: "10.0.0.0/24,192.168.1.100" <3>
+  csi.storage.k8s.io/provisioner-secret-name: manila-csi-secret
+  csi.storage.k8s.io/provisioner-secret-namespace: openshift-manila-csi-driver
+  csi.storage.k8s.io/controller-expand-secret-name: manila-csi-secret
+  csi.storage.k8s.io/controller-expand-secret-namespace: openshift-manila-csi-driver
+  csi.storage.k8s.io/node-stage-secret-name: manila-csi-secret
+  csi.storage.k8s.io/node-stage-secret-namespace: openshift-manila-csi-driver
+  csi.storage.k8s.io/node-publish-secret-name: manila-csi-secret
+  csi.storage.k8s.io/node-publish-secret-namespace: openshift-manila-csi-driver
+allowVolumeExpansion: true
+----
++
+<1> A descriptive name for your custom storage class.
+<2> The Manila share type. This should match an existing share type in your OpenStack environment.
+<3> Comma-separated list of IP addresses or CIDR subnets that are allowed to access the NFS shares. In this example, access is restricted to the `10.0.0.0/24` subnet and the specific IP address `192.168.1.100`.
+
+. Apply the storage class:
++
+[source,terminal]
+----
+$ oc apply -f custom-manila-storageclass.yaml
+----
+
+. Verify that the storage class was created:
++
+[source,terminal]
+----
+$ oc get storageclass csi-manila-gold-restricted
+----
+
+. Create a PVC that uses the custom storage class:
++
+.pvc-manila-restricted.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-manila-restricted
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: csi-manila-gold-restricted <1>
+----
++
+<1> Reference to your custom storage class with restricted access.
+
+. Apply the PVC:
++
+[source,terminal]
+----
+$ oc apply -f pvc-manila-restricted.yaml
+----
+
+[NOTE]
+====
+The `nfs-ShareClient` parameter accepts various formats:
+
+* Single IP address: `192.168.1.100`
+* CIDR subnet: `10.0.0.0/24`
+* Multiple entries: `10.0.0.0/24,192.168.1.100,172.16.0.0/16`
+
+Ensure that the specified IP addresses or subnets include the {product-title} cluster nodes to allow proper mounting of the persistent volumes.
+====
+
+[IMPORTANT]
+====
+When using custom storage classes with restricted access rules, ensure that:
+
+* The specified IP addresses or subnets include all {product-title} nodes that need to access the storage.
+* The Manila service in {rh-openstack} supports the share type specified in the storage class.
+* Network connectivity exists between the allowed clients and the Manila share servers.
+====
```

Comment:
I'll likely break this out into an additional little module. Very good start though!